src/nn.h

```diff
@@ -387,9 +401,35 @@ NN_EXPORT int nn_device (int s1, int s2);
 
 NN_EXPORT int nn_tcpmuxd (int port);
 
+/******************************************************************************/
+/*  Statistics.                                                               */
+/******************************************************************************/
+
+/*  Transport statistics  */
+#define NN_STAT_ESTABLISHED_CONNECTIONS 101
+#define NN_STAT_ACCEPTED_CONNECTIONS    102
+#define NN_STAT_DROPPED_CONNECTIONS     103
+#define NN_STAT_BROKEN_CONNECTIONS      104
+#define NN_STAT_CONNECT_ERRORS          105
+#define NN_STAT_BIND_ERRORS             106
+#define NN_STAT_ACCEPT_ERRORS           107
+
+#define NN_STAT_CURRENT_CONNECTIONS     201
+#define NN_STAT_INPROGRESS_CONNECTIONS  202
+#define NN_STAT_CURRENT_EP_ERRORS       203
+
+/*  The socket-internal statistics  */
+#define NN_STAT_MESSAGES_SENT           301
+#define NN_STAT_MESSAGES_RECEIVED       302
+#define NN_STAT_BYTES_SENT              303
+#define NN_STAT_BYTES_RECEIVED          304
+/*  Protocol statistics  */
+#define	NN_STAT_CURRENT_SND_PRIORITY    401
+
+NN_EXPORT NN_UINT64 nn_get_statistic (int s, int stat);
```

Comment:
I don't disagree with getting stats one at a time, but to get ahead of potential question -- why did you decided that statistics can only be retrieved one at a time?

What would the developer experience look like to be able to iterate through all known statistics (without needing to maintain a list of constants to the available stats)? Is there such a facility to discover all available statistics?